android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -595,6 +598,59 @@ class OkHttpTest {
     }
   }
 
+  @Test
+  fun testCachedRequest() {
+    assumeNetwork()
+
+    val ctxt = InstrumentationRegistry.getInstrumentation().targetContext.applicationContext
+
+    val cacheSize = 1L * 1024 * 1024 // 1MB
+    val cache = Cache(ctxt.cacheDir.resolve("testCache"), cacheSize)
+
+    try {
+      client = client.newBuilder()
+          .cache(cache)
+          .build()
+
+      val request = Request.Builder()
+          .url("https://httpbin.org/cache/3")
```

Comment:
use MockWebServer here?

Reply:
I've previously preferred real world servers to confirm that things are working end to end.  But given the tests are relatively stable, it's probably a good point to switch from manually running to running in Github.

Will copy the approach here https://twitter.com/colinwhi/status/1244426550918823937

Reply:
Hell yes. Yay @colinrtwhite !

Reply:
Nice! Glad this was useful for y'all 😄